lib/puppet/provider/keycloak_client/kcadm.rb

```diff
@@ -221,6 +242,19 @@ def create
         raise Puppet::Error, "kcadm delete clients/#{resource[:id]}/optional-client-scopes/#{scope_id}: #{e.message}"
       end
     end
+    role = nil
+    if resource[:roles]
+      roles = @property_hash[:roles] ||= get_client_roles(resource[:realm], resource[:id])
```

Comment:
I think my previous recommendation was wrong.  `@property_hash` won't be set during `create` because `create` is called when something does not exist.  So your previous approach was correct, only do this:

```
role = get_client_roles(resource[:realm], resource[:id])
```